File: lib/layer2_core/light/e_on_error_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnErrorLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl42;
      case 'standard': return cstl42;
      case 'blue': return cbl42;
      case 'green': return cgl42;
      case 'red': return crl42;
      case 'nothing': return cnol42;
      default: return cstl42;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnErrorLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnErrorLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl42
            AppTheme.STANDARD -> R.color.cstl42
            AppTheme.BLUE -> R.color.cbl42
            AppTheme.GREEN -> R.color.cgl42
            AppTheme.RED -> R.color.crl42
            AppTheme.NOTHING -> R.color.cnol42
        }
    }
}
